### ISSUE TITLE
Issue 02: coarse config と checkpoint metadata を trace heads 対応にする

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#363]
# Issue 02: coarse config と checkpoint metadata を trace heads 対応にする

## 背景

Issue 01 で coarse model が multi-task 出力を返せるようになる。次に、YAML config から pickability / usability head と loss 設定を読み、checkpoint metadata に保存・検証できるようにする。

現状は `model.out_chans == 1` と `ckpt.output_ids == ("P",)` が固定で、inference 側の checkpoint validation も `output_ids == ["P"]` を要求している。

## ゴール

以下を config で指定できるようにする。

```yaml
model:
  backbone: resnet18
  in_chans: 3
  out_chans: 1
  # 既存 encdec2d 設定はそのまま
  trace_heads:
    pickability:
      enabled: true
    coarse_usable:
      enabled: false
    hidden_chans: 64
    dropout: 0.0

train:
  pickability_pu:
    enabled: true
    weight: 0.1
    class_prior: 0.6
    beta: 0.0
    gamma: 1.0
  coarse_usability:
    enabled: false
    weight: 0.2
    label_meta_key: coarse_usable_label_view
    ignore_value: -1

infer:
  physics_input_score:
    mode: pickability_times_pmax
```

既存 config では `trace_heads` が無いため、完全に legacy と同じ挙動にする。

## 変更対象の候補

- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/config.py`
- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/build_model.py`
- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/train.py`
- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/infer.py`
- 必要なら examples:
  - `examples/config_train_fbpick_coarse.yaml`
  - `examples/config_infer_fbpick_coarse.yaml`

### 1. dataclass を追加する

`coarse/config.py` に次を追加する。

```python
@dataclass(frozen=True)
class CoarseTraceHeadCfg:
    pickability_enabled: bool
    coarse_usable_enabled: bool
    hidden_chans: int
    dropout: float

@dataclass(frozen=True)
class CoarsePickabilityPUCfg:
    enabled: bool
    weight: float
    class_prior: float | None
    beta: float
    gamma: float

@dataclass(frozen=True)
class CoarseUsabilityTrainCfg:
    enabled: bool
    weight: float
    label_meta_key: str
    ignore_value: int

@dataclass(frozen=True)
class CoarsePhysicsInputScoreCfg:
    mode: str
```

既存の `CoarseTrainCfg` / `CoarseInferRuntimeCfg` などに必要なフィールドを追加する。

### 2. config parse のデフォルト

デフォルトは必ず後方互換にする。

```yaml
model.trace_heads.pickability.enabled: false
model.trace_heads.coarse_usable.enabled: false
train.pickability_pu.enabled: false
train.coarse_usability.enabled: false
infer.physics_input_score.mode: pmax
```

`infer.physics_input_score.mode` の候補:

- `pmax`: legacy。`physics_input_score = coarse_pmax`
- `pickability_times_pmax`: Step 2 用
- `pickability_times_usability`: Step 3 用
- `pickability_times_usability_times_pmax`: 比較用に許可してもよい

### 3. model_sig を versioned にする

multi-task 有効時は `model_sig` を次の形式にする。

```python
{
    "encdec2d": encdec2d_kwargs,
    "trace_heads": {
        "pickability": {"enabled": True},
        "coarse_usable": {"enabled": False},
        "hidden_chans": 64,
        "dropout": 0.0,
    },
}
```

legacy config では、既存 checkpoint 互換のために flat `encdec2d_kwargs` のままでもよい。Codex 実装時は以下のどちらかに統一すること。

- A案: legacy は flat のまま、multi-task だけ nested。
- B案: すべて nested にし、checkpoint validation で legacy flat も受ける。

既存 checkpoint を壊しにくい A案を推奨する。

### 4. checkpoint metadata を拡張する

`coarse/train.py` の `ckpt_extra` に以下を追加する。

```python
"coarse_trace_heads": {
    "pickability_enabled": ...,
    "coarse_usable_enabled": ...,
},
"output_ids": ["P", "pickability", "coarse_usable"]  # 有効なものだけ
```

legacy では今まで通り `output_ids == ["P"]`。

### 5. checkpoint validation を更新する

`coarse/infer.py::validate_checkpoint_for_global_anchor_infer()` を更新する。

要件:

- legacy checkpoint `output_ids == ["P"]` は引き続き許可。
- infer config が pickability head を要求している場合、checkpoint に `pickability` が無ければエラー。
- infer config が coarse usable head を要求している場合、checkpoint に `coarse_usable` が無ければエラー。
- `softmax_axis == "time"` は `P` に対する意味として維持する。

## 受け入れ条件

- 既存 coarse train/infer config は変更なしで load できる。
- `model.trace_heads.pickability.enabled: true` を指定すると、typed config と `model_sig` に反映される。
- invalid な `hidden_chans <= 0`, `dropout < 0 or >= 1`, `weight < 0`, `class_prior not in (0,1)` で明確な例外が出る。
- infer config が `pickability_times_pmax` を要求し、checkpoint に pickability head が無い場合はエラーになる。
- legacy checkpoint は legacy infer config で通る。

## テスト

新規または既存テストに追加:

- `tests/test_fbpick_coarse_config_trace_heads.py`

確認内容:

1. legacy config parse。
2. pickability enabled config parse。
3. usability enabled config parse。
4. checkpoint validation: legacy OK。
5. checkpoint validation: required head missing で NG。

## 非ゴール

- loss の計算は Issue 03 / Issue 08。
- inference NPZ 保存は Issue 04。
